### PR TITLE
Matter: update sharing from Google|Apple Home to reflect current software

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Matter: update procedures on sharing from Google or Apple Home to reflect current software
- used iOS version 17.3

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/matter.markdown [hunk 1]
```diff
@@ -149,7 +149,7 @@ This will use the Bluetooth connection of your phone to add the device.
 
 ### To add a new device using the Android Companion app
 
-This will use the Bluetooth connection of your phone to add the device.
+This guide describes how to add a new device that has never been added to your network before. This will use the Bluetooth connection of your phone to add the device.
```
Comment (marcelveldt) [inline source/_integrations/matter.markdown]: is this clear enough ? Or do you want to refer to "just unpacked from the box" ?

Comment (c0ffeeca7) [reply source/_integrations/matter.markdown]: Well, the device I used wasn't really new new. It was just no longer paired to any server and freshly reset.

File: source/_integrations/matter.markdown [hunk 2]
```diff
@@ -173,29 +173,56 @@ This will use the Bluetooth connection of your phone to add the device.
...
+
+- Latest version of the Home Assistant Companion App installed.
+- Have the Matter device integrated to Apple Home or Google Home respectively
+- In Home Assistant, have the Matter integration installed.
+  - Go to {% my integrations title="**Settings** > **Devices & services**" %}.
+  - Add the **Matter (BETA)** integration.
+  - When prompted to **Select the connection method**:
+    - If you run Home Assistant OS in a regular setup: select **Submit**.
+      - This will install the official Matter server add-on.
+    - If you are already running the Matter server in another add-on, in or a custom container:
```
Comment (marcelveldt) [inline source/_integrations/matter.markdown]:
```suggestion
    - If you are running the Matter server in a custom container (not recommended):
```

File: source/_integrations/matter.markdown [hunk 3]
```diff
@@ -173,29 +173,56 @@ This will use the Bluetooth connection of your phone to add the device.
...
-4.  Choose Home Assistant from the list, you are redirected to the Home Assistant Companion app now. Press **Add device**.
-5.  Your device will now be added to Home Assistant. When the process finishes, you're redirected to the device page in Home Assistant.
+1. In Google Home, open the **Devices** page and select the cogwheel.
+2. On the **Settings** page, select your Matter device.
+3. Select **Linked Matter apps and services**.
+4. Select **Link apps and services** to link the device to Home Assistant.
+5. Choose Home Assistant from the list.
+   - You are redirected to the Home Assistant Companion app now.
+   - Select **Add device**.
+   - **Troubleshooting**: If Home Assistant fails to add the device, check if you have the Matter integration installed.
```
Comment (marcelveldt) [inline source/_integrations/matter.markdown]:
```suggestion
   - **Troubleshooting**: If Home Assistant fails to add the device, check if you have the Matter integration installed and the latest version of the Companion app.
```

**Review by marcelveldt — DISMISSED** [review]
2 small optional comments

**Review by marcelveldt — DISMISSED** [review]

**Review by marcelveldt — APPROVED** [review]

**Pull request merged**